pkg/daemon/ceph/client/filesystem.go

```diff
@@ -97,7 +99,10 @@ func GetFilesystem(context *clusterd.Context, clusterName string, fsName string)
 	args := []string{"fs", "get", fsName}
 	buf, err := NewCephCommand(context, clusterName, args).Run()
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get file system %s", fsName)
+		// if filesystem does not exist, return empty buf and error
+		if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
+			return nil, errors.Wrapf(err, "failed to get file system %s", fsName)
```

Comment:
If the filesystem doesn't exist we want to continue with no error. Here we need to return the err without wrapping it so that the caller can check for ENOENT and continue in that case. As you have it, the reconcile will still fail if it doesn't exist.
```suggestion
			return nil, err
```

Reply:
Ok, So that means if `ENOENT` is `2`, return without wrapping otherwise wrap the error and return as usual.

Reply:
I need to learn more about errors, it's a little difficult to get my head around handling errors.

Reply:
I'm suggesting the following... First, always return `err` from this method if it failed.
```
	args := []string{"fs", "get", fsName}
	buf, err := NewCephCommand(context, clusterInfo, args).Run()
	if err != nil {
		return err
	} 
```

Second, in the code where the filesystem is deleted, check for the error code if it doesn't exist. **It's up to the caller to decide if it's an error that the filesystem doesn't exist.**
```
func deleteFilesystem(...) error {
	filesystem, err := client.GetFilesystem(context, clusterInfo, fs.Name)
	if err != nil {
            if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
                 // If we're deleting the filesystem anyway, ignore the error that the filesystem doesn't exist
                 return nil
            }
	    return errors.Wrapf(err, "failed to get filesystem %q", fs.Name)
	}
```

Reply:
I see, this is completely different from what I was thinking. 
Thanks for the suggestion. I have made above changes along with the comment and pushing them in some time.

Reply:
> I'm suggesting the following... First, always return `err` from this method if it failed.
> 
> ```
> 	args := []string{"fs", "get", fsName}
> 	buf, err := NewCephCommand(context, clusterInfo, args).Run()
> 	if err != nil {
> 		return err
I believe this will be `nil` and `err` right?.Because we have to return 2 values. Right?.
> 	} 
> ```
> 
> Second, in the code where the filesystem is deleted, check for the error code if it doesn't exist. **It's up to the caller to decide if it's an error that the filesystem doesn't exist.**
> 
> ```
> func deleteFilesystem(...) error {
> 	filesystem, err := client.GetFilesystem(context, clusterInfo, fs.Name)
> 	if err != nil {
>             if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
>                  // If we're deleting the filesystem anyway, ignore the error that the filesystem doesn't exist
>                  return nil
>             }
> 	    return errors.Wrapf(err, "failed to get filesystem %q", fs.Name)
> 	}
> ```

Reply:
yes, correct

Reply:
Alright, made the changes accordingly.